### PR TITLE
Make index.addAll use status to increase performance

### PR DESCRIPTION
**Pull request by maxkorp**
See here: https://github.com/libgit2/libgit2/issues/2687
Doing this speeds things up from around 2500-3500ms for a small-medium repo to around 180-200ms. We should remove this once it's fixed in libgit2, but until then, this seems silly not to add. 

This also adds pathspec.create (since we need that to actually compare pathspecs)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -19,7 +21,30 @@ Index.prototype.entries = function() {
...
+    paths.push(path);
+  };
+  var idx = this;
+  return Pathspec.create(pathspec || "*")
+    .then(function(ps) {
+      pathspec = ps;
+      return Status.foreach(repo, statusCB);
+    })
+    .then(function() {
+      return Promise.resolve(paths);
```
Comment (tbranyen) [inline lib/index.js]: Can this just be: `return paths;`?

Comment (johnhaley81) [inline lib/index.js]: Yeah. I don't know the purpose of the `Promise.resolve`

**Comment by johnhaley81** [comment]
Are we ok with putting this in 0.3.0 milestone?

**Comment by tbranyen** [comment]
Works for me.

**Comment by johnhaley81** [comment]
I'm going to take out that `Promise.resolve(paths)` thing first.

**Pull request merged**